parquet-variant-compute/src/variant_get/output/variant.rs

```diff
@@ -18,11 +18,52 @@
 use crate::variant_get::output::OutputBuilder;
 use crate::{VariantArray, VariantArrayBuilder};
 use arrow::array::{Array, ArrayRef, AsArray, BinaryViewArray};
-use arrow::datatypes::Int32Type;
+use arrow::datatypes::{
+    Float16Type, Float32Type, Float64Type, Int16Type, Int32Type, Int64Type, Int8Type, UInt16Type,
+    UInt32Type, UInt64Type, UInt8Type,
+};
 use arrow_schema::{ArrowError, DataType};
 use parquet_variant::{Variant, VariantPath};
 use std::sync::Arc;
 
+macro_rules! cast_perfectly_shredded_primitive {
+    ($typed_value:expr, $variant_array:expr, $arrow_type:ty, $data_type:expr) => {{
+        let mut array_builder = VariantArrayBuilder::new($variant_array.len());
+        let primitive_array = $typed_value.as_primitive::<$arrow_type>();
+        for i in 0..$variant_array.len() {
+            if primitive_array.is_null(i) {
+                array_builder.append_null();
+            } else {
+                let value = primitive_array.value(i);
+                array_builder.append_variant(Variant::from(value));
+            }
+        }
```

Comment:
if https://github.com/apache/arrow-rs/pull/8105/files#diff-ed469f33b865066aa56a96aa5279ce67d4bf85ea34c7a8c12ffb16afef107b6fR23-R31 merges, this simplifies to:
```suggestion
        primitive_conversion_array!($arrow_type, $typed_value, array_builder);
```
TBD whether it's worth defining the macro just to turn 3 lines into one?

Reply:
Yeah, re-using that macro after it merges would work.